### PR TITLE
add (h0,cosi)<->(aplus,across) functions to utils.converting

### PR DESCRIPTION
**Pull request by dbkeitel**
Extracts the existing one from an `snr` internal method to `utils.converting`, and adds the inverse, so they can be reused by other modules too.

Inverse should match https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/_compute_fstatistic__v2_8c_source.html#l02045

### CODE REVIEW

File: pyfstat/utils/converting.py [hunk 1]
```diff
@@ -107,3 +107,62 @@ def gps_to_datestr_utc(gps):
...
+    """
+
+    if np.abs(aCross) > np.abs(aPlus):
+        raise ValueError("not valid for abs(aCross)>abs(aPlus)")
+
+    h0 = aPlus + np.sqrt(aPlus**2 - aCross**2)
+    if h0 > 0:
+        cosi = aCross / h0
+    else:
+        cosi = 0
```
Comment (Rodrigo-Tenorio) [inline pyfstat/utils/converting.py:167]: @dbkeitel what am I missing here? `h0 > 0` by definition?

```suggestion
    cosi = aCross / h0
```

**Review by Rodrigo-Tenorio — CHANGES_REQUESTED** [review]
Nice @dbkeitel .

I've put some suggestions so this functions can be used with numpy arrays as well.

File: pyfstat/utils/converting.py [hunk 2]
```diff
@@ -107,3 +107,62 @@ def gps_to_datestr_utc(gps):
...
+    Returns
+    -------
+    h0: float
+        Nominal GW amplitude.
+    cosi: float
+        Cosine of the source inclination w.r.t. line of sight.
+    """
+
+    if np.abs(aCross) > np.abs(aPlus):
+        raise ValueError("not valid for abs(aCross)>abs(aPlus)")
```
[the remark this answers is not in the document]
Comment (Rodrigo-Tenorio) [reply pyfstat/utils/converting.py:161]: Right now it fails, we could also decide to return `NaN` on those invalid values, if you prefer.

Thoughs?

File: pyfstat/utils/converting.py (hunk 1 above)
Comment (dbkeitel) [reply pyfstat/utils/converting.py:167]: No, only `h0 >= 0` is guaranteed. ;)

Turning around the if-else, so the special case is specifically labelled `==0`, would make it more intuitive, but this order is inherited from CFSv2.

File: pyfstat/utils/converting.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rodrigo-Tenorio) [reply pyfstat/utils/converting.py:161]: > ? Seems more intuitive to read to me than the fraction version, and doesn't have a singularity..?

Didn't think to much about it, I actually like yours better :+1:.

File: pyfstat/utils/converting.py (hunk 1 above)
Comment (Rodrigo-Tenorio) [reply pyfstat/utils/converting.py:167]:
```suggestion
    cosi = np.divide(aCross, h0, out=np.zeros_like(aCross), where=(h0 > 0))
```

**Review by Rodrigo-Tenorio — APPROVED** [review]

**Pull request merged**